bundle pack --with-semantics: composePreviewDaemonStart failure aborts the whole pack (graceful degradation is unreachable)

## Summary

`compose-preview bundle pack --with-semantics` (added in #1845, for #1843) is documented as best-effort: a missing descriptor / daemon open / render failure / unsupported backend should warn to stderr and leave the already-written bundle valid rather than failing the pack. In practice, on the **desktop (CMP/Skiko) backend**, a failure in the daemon-start phase fails the **entire pack** with `Gradle bundle task failed` (exit 1) — the bundle is never written and the graceful-degradation path never runs.

Observed on `compose-preview` **0.15.5**.

## Root cause

In `cli/.../BundleCommand.kt`, `--with-semantics` appends `composePreviewDaemonStart` to the **same Gradle invocation** as the render+bundle:

```kotlin
val tasks = buildList {
    if (!noRender) add(":${target.gradlePath}:composePreviewRender")
    add(":${target.gradlePath}:composePreviewBundle")
    if (packSemantics) add(":${target.gradlePath}:composePreviewDaemonStart")   // same runGradle call
  }.toTypedArray()
val ok = runGradle(gradle, *tasks, arguments = gradleArgsWithForce(gradleArgs))
if (!ok) {
  System.err.println("Gradle bundle task failed.")
  exitProcess(1)                                   // <-- aborts here
}
...
val semanticsLine = if (packSemantics) packSemanticsBlob(target, resolvedOutput, meta) else null
```

The graceful-degradation logic (`DescriptorMissing` / `OpenFailed` / `Ok`-but-empty → "bundle written without semantics") all lives in `packSemanticsBlob` / `DaemonSemanticsFetcher`, which only run **after** that `runGradle` returns success. When `composePreviewDaemonStart` itself fails at the Gradle level, `ok == false` and the command `exitProcess(1)`s at the `Gradle bundle task failed` line — `packSemanticsBlob` is unreachable, so the "best-effort" contract doesn't hold. The `DaemonSemanticsFetcher` try/catch guards the *render/open* phase but not the *daemon-start* Gradle task.

## Reproduction

In a Compose Multiplatform module that renders on the desktop backend (here `:meshcore-components`, an Android KMP library with a `jvm("desktop")` target, `backend=desktop`):

```bash
compose-preview bundle pack --module meshcore-components \
  --id <previewId>... --with-semantics \
  -o build/design-parity/candidates.bundle.png --timeout 600
```

The same command **without** `--with-semantics` succeeds and produces a valid (visual/structural) bundle.

### Actual

```
FAILURE: Build failed with an exception.
* What went wrong:
* Try:
> Run with --scan to get full insights from a Build Scan (powered by Develocity).

Run with --verbose for full build output.
Gradle bundle task failed.
##[error]Process completed with exit code 1.
```

(~36s in; the underlying Gradle exception detail is swallowed because the CLI doesn't pass `--verbose`/`--scan` through here — see the secondary issue below.)

### Expected

Per #1845's contract: the bundle is still written (visual/structural entries intact, polyglot valid), `--with-semantics` warns to stderr that it couldn't capture semantics on this backend, and the pack exits **0**.

## Suggested fix

Don't let a `composePreviewDaemonStart` failure abort the pack. Options:

1. Run `composePreviewDaemonStart` (and the semantics fetch) in a **separate** Gradle invocation **after** the render+bundle invocation succeeds, and treat its failure the same as the other best-effort outcomes (warn, leave the bundle, exit 0).
2. Or, if it must stay in one invocation, distinguish *which* task failed and only `exitProcess(1)` when render/bundle failed — fall through to the best-effort semantics handling when only daemon-start failed.

## Secondary

When the daemon-start/bundle Gradle task fails, the CLI prints `Gradle bundle task failed` but swallows the Gradle exception (it suggests `--verbose`/`--scan` but doesn't pass them through), so the *root* reason for the desktop `composePreviewDaemonStart` failure isn't visible from the pack output. Surfacing the underlying Gradle error (or auto-passing `--verbose` on failure) would make this diagnosable. There may additionally be a genuine desktop daemon-start defect underneath — but it's masked until the abort + swallowed-output issues above are addressed.

## Impact / context

This is consumed by `yschimke/meshcore-mobile`'s `design-parity.yml`, which packs the candidate on the desktop backend. Adding `--with-semantics` (to light up design-parity's token/contrast/a11y checks, which otherwise report "missing from candidate" per #1843) turned the workflow red and left its published report stale. Worked around there by dropping `--with-semantics` (back to visual-only) until this is fixed.

--- a/cli/src/main/kotlin/ee/schimke/composeai/cli/BundleCommand.kt
+++ b/cli/src/main/kotlin/ee/schimke/composeai/cli/BundleCommand.kt
@@ -174,11 +174,13 @@ private class PackSubcommand(private val args: List<String>) {
             }
             // `--with-semantics` carries the per-preview semantics blob (issue #1843). The
             // semantics tree is produced exclusively by the daemon (the standalone
-            // composePreviewRender task writes no semantics sidecars), so we also start the daemon
-            // in the same Gradle invocation — its launch descriptor is written fresh against the
-            // consumer's current classpath, then a short-lived render session reads the blobs back.
-            // Pointless without a render (the daemon needs something to capture), so skip when
-            // --no-render.
+            // composePreviewRender task writes no semantics sidecars), so we start the daemon and
+            // read the blobs back below. composePreviewDaemonStart runs in its OWN Gradle
+            // invocation (not bundled with render+bundle), AFTER the bundle is already written, so
+            // a
+            // daemon-start failure degrades gracefully instead of aborting the pack —
+            // `--with-semantics` is best-effort (issue #1885). Pointless without a render (the
+            // daemon needs something to capture), so skip when --no-render.
             val packSemantics = withSemantics && !noRender
             if (withSemantics && noRender) {
               System.err.println(
@@ -189,12 +191,15 @@ private class PackSubcommand(private val args: List<String>) {
               buildList {
                   if (!noRender) add(":${target.gradlePath}:composePreviewRender")
                   add(":${target.gradlePath}:composePreviewBundle")
-                  if (packSemantics) add(":${target.gradlePath}:composePreviewDaemonStart")
                 }
                 .toTypedArray()
             val ok = runGradle(gradle, *tasks, arguments = gradleArgsWithForce(gradleArgs))
             if (!ok) {
-              System.err.println("Gradle bundle task failed.")
+              System.err.println(
+                "Gradle bundle task failed." +
+                  if (!verbose) " Re-run with --verbose to surface the underlying Gradle error."
+                  else ""
+              )
               exitProcess(1)
             }
 
@@ -216,8 +221,31 @@ private class PackSubcommand(private val args: List<String>) {
               }
             // Inject semantics (if requested) before the summary so its byte count reflects the
             // enriched bundle; the summary line for semantics is printed after the main summary.
+            // The bundle is already written and valid at this point, so everything below is
+            // best-effort: a daemon-start / open / render failure warns and leaves the bundle as-is
+            // (issue #1885), never failing the pack.
             val semanticsLine =
-              if (packSemantics) packSemanticsBlob(target, resolvedOutput, meta) else null
+              if (packSemantics) {
+                // Start the daemon in a SEPARATE Gradle invocation — its failure must NOT abort the
+                // pack. The launch descriptor is regenerated fresh against the consumer's current
+                // classpath, then DaemonSemanticsFetcher reads the blobs back.
+                val daemonStarted =
+                  runGradle(
+                    gradle,
+                    ":${target.gradlePath}:composePreviewDaemonStart",
+                    arguments = gradleArgsWithForce(gradleArgs),
+                  )
+                if (!daemonStarted) {
+                  System.err.println(
+                    "bundle pack: --with-semantics could not start the preview daemon for " +
+                      "${target.gradlePath} (composePreviewDaemonStart failed) — bundle written " +
+                      "without semantics. Re-run with --verbose to surface the underlying Gradle error."
+                  )
+                  null
+                } else {
+                  packSemanticsBlob(target, resolvedOutput, meta)
+                }
+              } else null
 
             printPackSummary(resolvedOutput, meta)
             semanticsLine?.let { println(it) }

